### PR TITLE
update async commit related documentation (#5114)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #5114 to release-5.0
You can switch your code base to this Pull Request by using [git-extras](https://github.com/tj/git-extras):
```bash
# In docs repo:
git pr https://github.com/pingcap/docs/pull/5131
```

After apply modifications, you can push your change to this PR via:
```bash
git push git@github.com:ti-srebot/docs.git pr/5131:release-5.0-8750fd74ea40
```

---

### What is changed, added or deleted? (Required)

TiDB 5.0 GA will not contain configurations related to async commit or 1PC. So they are removed.

And we also introduce a weaker consistency model for transactions. Add documentation about it.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/5607
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by youjiali1995 — COMMENTED** [review]
/lgtm

File: system-variables.md [hunk 1]
```diff
@@ -361,23 +361,25 @@ Constraint checking is always performed in place for pessimistic transactions (d
 >
 > Currently, this feature is incompatible with TiDB Binlog in some scenarios and might cause semantic changes on a transaction. For more usage precautions of this feature, refer to [Incompatibility issues about transaction semantic](https://github.com/pingcap/tidb/issues/21069) and [Incompatibility issues about TiDB Binlog](https://github.com/pingcap/tidb/issues/20996).
 
-### tidb_enable_async_commit <span class="version-mark">New in v5.0.0-rc</span>
+### tidb_enable_async_commit <span class="version-mark">New in v5.0 RC</span>
 
-> **Warning:**
+- Scope: SESSION | GLOBAL
+- Default value: `ON` for newly created clusters. If your cluster before upgrade was v5.0 RC or later, the variable value stays unchanged. If your cluster before upgrade was v4.0 or earlier, the variable value defaults to `OFF` after the upgrade.
```
Comment (TomShawn) [inline system-variables.md:367]:
```suggestion
- Default value: For newly created clusters, the default value of v5.0 RC is `OFF` and the default value of v5.0 GA or later is `ON`. If your cluster was upgraded to v5.0 GA from v5.0 RC, the variable value stays unchanged. If your cluster was upgraded to v5.0 GA from v4.0 or earlier, the variable value defaults to `OFF` after the upgrade.
```

File: system-variables.md [hunk 2]
```diff
@@ -361,23 +361,25 @@ Constraint checking is always performed in place for pessimistic transactions (d
...
 
-> - This feature is incompatible with [TiCDC](/ticdc/ticdc-overview.md) and might cause TiCDC to run abnormally.
-> - This feature is incompatible with [Compaction Filter](/tikv-configuration-file.md#enable-compaction-filter-new-in-v500-rc). If you use the two features at the same time, write loss might occur.
-> - This feature is incompatible with TiDB Binlog and does not take effect when TiDB Binlog is enabled.
+### tidb_enable_1pc <span class="version-mark">New in v5.0 RC</span>
 
 - Scope: SESSION | GLOBAL
-- Default value: OFF
-- This variable controls whether to enable the async commit feature for the second phase of the two-phase transaction commit to perform asynchronously in the background. Enabling this feature can reduce the latency of transaction commit.
+- Default value: `ON` for newly created clusters. If your cluster before upgrade was v5.0 RC or later, the variable value stays unchanged. If your cluster before upgrade was v4.0 or earlier, the variable value defaults to `OFF` after the upgrade.
```
Comment (TomShawn) [inline system-variables.md:377]:
```suggestion
- Default value: For newly created clusters, the default value of v5.0 RC is `OFF` and the default value of v5.0 GA or later is `ON`. If your cluster was upgraded to v5.0 GA from v5.0 RC, the variable value stays unchanged. If your cluster is upgraded to v5.0 GA from v4.0 or earlier, the variable value defaults to `OFF` after the upgrade.
```

**Comment by TomShawn** [comment]
This PR aligns with https://github.com/pingcap/docs-cn/pull/5814.

**Pull request merged**